diff -ruNp lxqt-notificationd-orig/src/notificationarea.cpp lxqt-notificationd/src/notificationarea.cpp
--- lxqt-notificationd-orig/src/notificationarea.cpp	2025-10-28 16:32:05.000000000 +0330
+++ lxqt-notificationd/src/notificationarea.cpp	2025-11-06 19:49:22.035541456 +0330
@@ -223,6 +223,7 @@ void NotificationArea::setLayerShell()
                 }
                 layershell->setMargins(QMargins(m_spacing, m_spacing, m_spacing, m_spacing));
                 LayerShellQt::Window::Anchors anchors;
+                LayerShellQt::Window::Anchor edge = LayerShellQt::Window::AnchorTop;
                 if (QL1S("top-center") == m_placement)
                 {
                     anchors = {LayerShellQt::Window::AnchorTop};
@@ -238,24 +239,31 @@ void NotificationArea::setLayerShell()
                 else if (QL1S("center-left") == m_placement)
                 {
                     anchors = {LayerShellQt::Window::AnchorLeft};
+                    edge = LayerShellQt::Window::AnchorLeft;
                 }
                 else if (QL1S("center-right") == m_placement)
                 {
                     anchors = {LayerShellQt::Window::AnchorRight};
+                    edge = LayerShellQt::Window::AnchorRight;
                 }
                 else if (QL1S("bottom-left") == m_placement)
                 {
                     anchors = {LayerShellQt::Window::AnchorLeft | LayerShellQt::Window::AnchorBottom};
+                    edge = LayerShellQt::Window::AnchorBottom;
                 }
                 else if (QL1S("bottom-center") == m_placement)
                 {
                     anchors = {LayerShellQt::Window::AnchorBottom};
+                    edge = LayerShellQt::Window::AnchorBottom;
                 }
                 else if (QL1S("bottom-right") == m_placement)
                 {
                     anchors = {LayerShellQt::Window::AnchorBottom | LayerShellQt::Window::AnchorRight};
+                    edge = LayerShellQt::Window::AnchorBottom;
                 }
                 layershell->setAnchors(anchors);
+                // WARNING: Some compositors may demand an exclusive edge, although we need none.
+                layershell->setExclusiveEdge(edge);
             }
         }
     }
